Source/WebCore/html/CheckboxInputType.cpp

```diff
@@ -100,7 +110,7 @@ bool CheckboxInputType::matchesIndeterminatePseudoClass() const
 bool CheckboxInputType::shouldAppearIndeterminate() const
 {
     ASSERT(element());
-    return element()->indeterminate();
+    return element()->indeterminate() && !element()->isSwitch();
```

Comment:
Just to confirm my understanding here for why `indeterminate()` doesn't encapsulate `isSwitch()`: Do we still want to reflect `indeterminate` through JS, but not for rendering?

Reply:
Yeah, `indeterminate` state can always be changed regardless of the control. It's a bit weird, but it seems better to be consistent with the weirdness.